GVRf/Framework/jni/objects/scene_object.cpp

```diff
@@ -250,31 +250,217 @@ bool SceneObject::isColliding(SceneObject *scene_object) {
+    // A more efficient checking would be return three outcomes: 1) completely outside 2) completely inside 3) intersecting the frustum
+    // 1) will remain the same so that all children will be culled out
+    // 2) improves by directly counting in all its children without going over the frustum tests for each of its child
+    // 3) by further utilizing plane masking, is_cube_in_frustum() only needs to check the planes that intersect with the parent,
+    //    rather than all six planes for each of its child
+    is_inside = is_cube_in_frustum(frustum, bounding_volume_);
+
+    // Cull out the object and all its children if its bounding volume is completely outside the frustum
+    if (!is_inside) {
+        if (DEBUG_RENDERER) {
+            LOGD(
+                    "FRUSTUM: HBV not in frustum, cull out %s and all its children\n",
+                    name_.c_str());
+        }
+
+        return 0;
+    }
+
+    // 2. Skip the empty objects with no render data
+    if (render_data_ == NULL || render_data_->pass(0)->material() == NULL) {
+        if (DEBUG_RENDERER) {
+            LOGD("FRUSTUM: no render data skip %s\n", name_.c_str());
+        }
+
+        return 1;
+    }
+
+    // 3. Check if the object against the Level-of-details
+    // Transform the bounding sphere
+    glm::vec4 transformed_sphere_center(bounding_volume_.center(), 1.0f);
+
+    // Calculate distance from camera
+    glm::vec3 camera_position = camera->owner_object()->transform()->position();
+    glm::vec4 position(camera_position, 1.0f);
+    glm::vec4 difference = transformed_sphere_center - position;
+    float distance = glm::dot(difference, difference);
+
+    // this distance will be used when sorting transparent objects
+    render_data_->set_camera_distance(distance);
+
+    // if the object is not in the correct LOD level, cull out itself and all its children
+    if (!inLODRange(distance)) {
+        if (DEBUG_RENDERER) {
+            LOGD(
+                    "FRUSTUM: not in lod range, cull out %s and all its children\n",
+                    name_.c_str());
+        }
+
+        return 0;
+    }
+
+    // 4. Check if the object itself is intersecting with or inside the frustum
+    if (children_.size() > 0) {
+        is_inside = is_cube_in_frustum(frustum, mesh_bounding_volume);
+        //	is_inside = sphereInFrustum(frustum, mesh_bounding_volume);
+    }
+
+    // if the object is not in the frustum, cull out itself but continue testing its children
+    if (DEBUG_RENDERER) {
+        if (!is_inside) {
+            LOGD("FRUSTUM: mesh not in frustum, cull out %s\n", name_.c_str());
+        } else {
+            LOGD("FRUSTUM: mesh in frustum, render %s and all its children\n",
+                    name_.c_str());
+        }
+    }
+
+    return is_inside ? 2 : 1;
+}
+
+bool SceneObject::is_cube_in_frustum(float frustum[6][4],
```

Comment:
It doesn't need to be const. It may be due to the compilation setting that complaining about the conversion, you can make a change to the code to make it compile or change the compilation setting.